ostd/src/arch/x86/timer/apic.rs

```diff
@@ -85,56 +106,53 @@ fn init_periodic_mode() -> IrqLine {
         apic.set_timer_init_count(0xFFFF_FFFF);
     });
 
-    static IS_FINISH: AtomicBool = AtomicBool::new(false);
-    static INIT_COUNT: AtomicU64 = AtomicU64::new(0);
-
     x86_64::instructions::interrupts::enable();
     while !IS_FINISH.load(Ordering::Acquire) {
         x86_64::instructions::hlt();
     }
     x86_64::instructions::interrupts::disable();
     drop(irq);
+}
 
-    // Init APIC Timer
-    let timer_irq = IrqLine::alloc().unwrap();
-
+fn init_periodic_mode(timer_irq: &IrqLine) {
     apic::with_borrow(|apic| {
         apic.set_timer_init_count(INIT_COUNT.load(Ordering::Relaxed));
         apic.set_lvt_timer(timer_irq.num() as u64 | (1 << 17));
         apic.set_timer_div_config(DivideConfig::Divide64);
     });
+}
 
-    return timer_irq;
-
-    fn pit_callback(trap_frame: &TrapFrame) {
-        static IN_TIME: AtomicU64 = AtomicU64::new(0);
-        static APIC_FIRST_COUNT: AtomicU64 = AtomicU64::new(0);
-        // Set a certain times of callbacks to calculate the frequency
-        const CALLBACK_TIMES: u64 = TIMER_FREQ / 10;
-
-        if IN_TIME.load(Ordering::Relaxed) < CALLBACK_TIMES || IS_FINISH.load(Ordering::Acquire) {
-            if IN_TIME.load(Ordering::Relaxed) == 0 {
-                let remain_ticks = apic::with_borrow(|apic| apic.timer_current_count());
-                APIC_FIRST_COUNT.store(0xFFFF_FFFF - remain_ticks, Ordering::Relaxed);
-            }
-            IN_TIME.fetch_add(1, Ordering::Relaxed);
-            return;
-        }
+static IS_FINISH: AtomicBool = AtomicBool::new(false);
+static INIT_COUNT: AtomicU64 = AtomicU64::new(0);
+
+fn pit_callback(trap_frame: &TrapFrame) {
```

Comment:
Please keep this method inside the `init_apic_freq` method. I don't see a valid reason to move it out.